modules/ml/include/opencv2/ml.hpp

```diff
@@ -1164,6 +1164,14 @@ class CV_EXPORTS_W RTrees : public DTrees
      */
     CV_WRAP virtual Mat getVarImportance() const = 0;
 
+    /** Returns the result of each individual tree in the forest.
+    In case the model is a regression problem, the method will return each of the tree's
+    result for each of the sample cases. If the model is a classifier, it will return 
+    a Mat with samples + 1 rows, where the first row gives the class number and the 
+    following rows return the votes each class had for each sample.
+    */
+    CV_WRAP virtual Mat predictProb(const InputArray inputs, int flags) const = 0;
```

Comment:
Also now we can't add pure virtual methods to public classes. It can break ABI compatibility. In current implementation we already have implicit definitions for this method in `DTrees` and `Rtrees`, so it can be just virtual.